docs/concepts/tools/kubectl/object-management-using-declarative-config.md

```diff
@@ -0,0 +1,517 @@
+- Selectors and PodTemplate Labels on any controller
+- Deployment RolloutStrategy
+
+### How to clear server defaulted fields or fields set by other writers
+
+As of Kubernetes 1.5, fields that do not appear in the configuration cannot be
+cleared.  For lists that are merged, this means elements of the list
+that do not appear in the configuration cannot be deleted.  The
+following workaround exist:
+
+Option 1: remove the field by directly editing the live object
+
+Option 2: remove the field through the configuration
+- Add the field to the configuration to match the live object
+- Apply the configuration - updates the annotation to include the field
+- Delete the field from the configuration
+- Apply the configuration - delete the field from the live object and annotation.
+
+{% comment %}
+TODO(1.6): Update this with the following for 1.6
+
+Fields that do not appear in the configuration can be cleared by
+setting their values to `null` and then applying the configuration.
+For fields defaulted by the server, this will trigger re-defaulting
+the values.
+{% endcomment %}
+
+## How to change ownership of a field between configuration and live writers
+
+Individual object fields should be owned by a single writer.
+
+### Change owner from a live writer to configuration
+
+Add the field to the configuration.  Stop the live writer from writing to the field.
+
+### Change owner from configuration to a live writer.
+
+As of Kubernetes 1.5, changing ownership of a field from configuration to
+a live writer requires manual steps.
+
+- Remove the field from the configuration
+- Remove the field from the `kubectl.kubernetes.io/last-applied-configuration` annotation on the live object.  This can be done using `edit`
+
+{% comment %}
+TODO(pwittrock): Update this once we support the following.
+
+- Remove the field from the configuration
+- Run `kubectl apply set last-applied-config -f <file>`
+{% endcomment %}
+
+## Migrating management methods
+
+Kubernetes objects should be managed using only one method at a time.
+Switching from one method to another is possible, but is a manual process.
+
+**Note:** The following exceptions:
+- using imperative delete with declarative management
+
+{% comment %}
+TODO(pwittrock): We need to make using imperative commands with
+declarative object configuration work so that it doesn't write the
+fields to the annotation, and instead.  Then add this bullet point.
+
+- using imperative commands with declarative configuration to manage where each manages different fields.
+{% endcomment %}
+
+### Migrating objects from imperative command management to declarative object configuration
+
+Migrating objects from imperative command management to declarative object
+configuration involves several manual steps.
+
+1. Export live object to a local object configuration file
+  - `kubectl get <kind>/<name> -o yaml --export > <kind>_<name>.yaml`
+2. Manually remove the status field from the object configuration in a text editor
```

Comment:
Not necessary, but it should be done.  The configuration file shouldn't include fields that it doesn't manage.